### PR TITLE
[Impersonation] Add User Impersonation UI option for Admins fixes #46

### LINKED ISSUE #46: [Impersonation] Add User Impersonation UI option for Admins
Impersonation is helpful feature for debugging in both dev and prod environments. The current implementation is such that, if the logged in user is an Admin, you can add a cookie named [iuid](https://github.com/coding-blocks/gondor/blob/master/api/middlewares/authenticate.js#L14) with the value of `user_id` of any user, on refresh you will be logged in as that user. Removing the cookie will take you back to your own account.

The idea is to create a UI option for to impersonate any user.
1) Create a modal which shows a [user select field](https://github.com/coding-blocks/gondor/blob/master/app/components/UserSelect/index.js), on submit it adds the cookie and reloads the page.

2) Add a dropdown item above logout button in the navbar named Impersonate, clicking on which the modal should open.
Screenshot 2020-05-16 at 4 50 05 PM

### PR DESCRIPTION
**Pull request by jugshaurya**
This PR will add the Impersonate UI and functionality for Admins and sets a cookie iuid to user_id and on deleting that cookie on refresh Admin is logged in as him/herself.

Demo: -
test

### CODE REVIEW

**Comment by jugshaurya** [comment]
@hereisnaman, Please review #77

File: app/components/UserSelect/index.js [hunk 1]
```diff
@@ -29,7 +29,10 @@ const UserSelect = ({
             : [value.id]
           : []),
       ],
-      availabilityDuring: availabilityStatusDuring,
+      availabilityDuring: availabilityStatusDuring || {
+        start_at: 0,
+        end_at: 0,
+      },
```
Comment (jugshaurya) [inline app/components/UserSelect/index.js:35]: **Hep here:** required to make changes here,  if want to hide the status in Impersonate dropdown then availabilityStatusDuring must be undefined and without {    start_at: 0,    end_at: 0, }, dropdown options were not coming.

File: app/modals/Impersonate/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  return (
+    <Modal isOpen={true} size="md">
+      <ModalHeader>Impersonate</ModalHeader>
+      <ModalBody>
+        <Form className="row">
+          <Form.Group className="has-float-label mb-4 col-12">
+            <Form.Label>Impersonate</Form.Label>
+            <UserSelect
+              name="impersonate"
```
Comment (0xnmn) [inline app/modals/Impersonate/index.js]: Make name and placeholder -> 'User'.

File: app/modals/Impersonate/index.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import { Modal, ModalHeader, ModalBody, ModalFooter, Button } from 'reactstrap';
+import Router from 'next/router';
+
+const Impersonate = ({ onClose }) => {
+  const [user, setUser] = useState('');
+
+  const handleImpersonate = () => {
+    const user_id = user.id;
+    document.cookie = `iuid=${user_id}`;
+    const { pathname } = Router;
```
Comment (0xnmn) [inline app/modals/Impersonate/index.js]: Use https://nextjs.org/docs/api-reference/next/router#userouter hook

**Review by 0xnmn — CHANGES_REQUESTED** [review]

**Comment by jugshaurya** [comment]
@hereisnaman, using next router hook, page is not refreshing. I tried.

n

**Comment by 0xnmn** [comment]
> @hereisnaman, using next router hook, page is not refreshing. I tried.
> 
> n

extract pathname from router and assign it to window.location.

**Comment by jugshaurya** [comment]
@hereisnaman changes done, review please

**Review by 0xnmn — APPROVED** [review]

**Pull request merged**